### PR TITLE
Possible fixes to reduce VS memory usage

### PR DESCRIPTION
**Pull request by dsyme**
@TIHan and others have noticed excessive creation of ILModuleReader.  Here's a possible fix, to be evaluated

### CODE REVIEW

**Comment by cartermp** [comment]
We should also have a tracking issue with concrete data on the number (8000+) of entities based on the dump file we've been working on cc @TIHan

File: src/absil/ilread.fs [hunk 1]
```diff
@@ -3601,11 +3602,11 @@ let openMetadataReader (fileName, mdfile: BinaryFile, metadataPhysLoc, peinfo, p
     let inbase = Filename.fileNameOfPath fileName + ": "
 
     // All the caches. The sizes are guesstimates for the rough sharing-density of the assembly 
-    let cacheAssemblyRef = mkCacheInt32 reduceMemoryUsage inbase "ILAssemblyRef" (getNumRows TableNames.AssemblyRef)
+    let cacheAssemblyRef = mkCacheInt32 false inbase "ILAssemblyRef" (getNumRows TableNames.AssemblyRef)
```
Comment (TIHan) [inline src/absil/ilread.fs]: Why not make everything "false" here?I feel like we should be caching everything.

File: src/absil/ilread.fs [hunk 2]
```diff
@@ -3964,10 +3965,15 @@ type ILModuleReaderImpl(ilModule: ILModuleDef, ilAssemblyRefs: Lazy<ILAssemblyRe
         member x.Dispose() = dispose()
     
 // ++GLOBAL MUTABLE STATE (concurrency safe via locking)
-type ILModuleReaderCacheLockToken() = interface LockToken
 type ILModuleReaderCacheKey = ILModuleReaderCacheKey of string * DateTime * ILScopeRef * bool * ReduceMemoryFlag * MetadataOnlyFlag
-let ilModuleReaderCache = new AgedLookup<ILModuleReaderCacheLockToken, ILModuleReaderCacheKey, ILModuleReader>(stronglyHeldReaderCacheSize, areSimilar=(fun (x, y) -> x = y))
-let ilModuleReaderCacheLock = Lock()
+
+// // Cache to extend the lifetime of readers that are eligible for GC
+// let ilModuleReaderCache1 = new ConcurrentDictionary<ILModuleReaderCacheKey, ILModuleReader>(HashIdentity.Structural)
```
Comment (TIHan) [inline src/absil/ilread.fs]: Is commenting this intentional?

File: src/absil/ilread.fs (hunk 1 above)
Comment (dsyme) [reply src/absil/ilread.fs]: Becasause
1. These caches themselves take space 
2. It's not clear it's worth it - if the items aren't being duplicated in long term storage the cache isn't worth it - so we should work based off evidence that duplication is happening.

File: src/absil/ilread.fs (hunk 2 above)
Comment (dsyme) [reply src/absil/ilread.fs]: See https://github.com/Microsoft/visualfsharp/pull/6390, which keeps both caches and is more conservative.

**Comment by KevinRansom** [comment]
@dsyme, how shall we choose between these two approaches?

**Comment by dsyme** [comment]
Closing this since I think #6390 is better

**Pull request closed without merging**